cfgov/core/middleware.py

```diff
@@ -99,3 +100,29 @@ def __call__(self, request):
         response = self.get_response(request)
         translation.deactivate()
         return response
+
+
+class SelfHealingMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        if response.status_code != 404:
```

Comment:
Just a little extra clarity:

```suggestion
        # If this request isn't 404ing, just return the existing response.
        if response.status_code != 404:
```